web/src/app/schedules/ScheduleRuleCreateDialog.js

```diff
@@ -25,20 +25,27 @@ const query = gql`
 
 export default function ScheduleRuleCreateDialog(props) {
   const { scheduleID, targetType, onClose } = props
+
+  const { data, ...queryStatus } = useQuery(query, {
+    variables: { id: scheduleID },
+  })
+
   const [value, setValue] = useState({
     targetID: '',
     rules: [
       {
-        start: DateTime.local().startOf('day').toUTC().toISO(),
-        end: DateTime.local().plus({ day: 1 }).startOf('day').toUTC().toISO(),
+        start: DateTime.local({ zone: data?.schedule.timeZone ?? 'UTC' })
+          .startOf('day')
+          .toISO(),
+        end: DateTime.local({ zone: data?.schedule.timeZone ?? 'UTC' })
+          .plus({ day: 1 })
+          .startOf('day')
+          .toISO(),
```

Comment:
UTC fallback won't work this way because once the real data loads, it won't update and would lead to subtle bugs.

But, since we are guaranteed to have the schedule info cached, we can probably just omit the check.

```suggestion
        start: DateTime.local({ zone: data.schedule.timeZone })
          .startOf('day')
          .toISO(),
        end: DateTime.local({ zone: data.schedule.timeZone })
          .plus({ day: 1 })
          .startOf('day')
          .toISO(),
```